### PR TITLE
Fix notcurses github action

### PR DESCRIPTION
**Pull request by Yaraslaut**
It looks like actions are working well now, even though nothing significant has been changed, apart from the logging level for these tests. Locally, I notice that notucrses-demo can fail from time to time due to events like terminal resizing. I think it is sufficient for us to have this PR merged at the moment and if needed to investigate further how to deal with it. for example to build notcurses from source inside actions as @christianparpart  few times proposed 

Closes https://github.com/contour-terminal/contour/issues/1402

### CODE REVIEW

File: src/vtbackend/Terminal.cpp [hunk 1]
```diff
@@ -263,12 +263,12 @@ bool Terminal::processInputOnce()
 
     auto const readResult = readFromPty();
 
-    if (!readResult)
+    if (!readResult.has_value())
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:266]: this should be equivalent. I am surprised and wondering why you did that. :)

File: src/vtbackend/Terminal.cpp [hunk 2]
```diff
@@ -263,12 +263,12 @@ bool Terminal::processInputOnce()
...
     {
+        terminalLog()("PTY read failed. {}", strerror(errno));
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:268]: that's a good thing to log, but maybe for everything except `EINTR` and `EAGAIN`. Maybe 2 lines below? Or was it intentionally placed above the if-condition?

File: src/vtbackend/Terminal.cpp (hunk 1 above)
Comment (Yaraslaut) [reply src/vtbackend/Terminal.cpp:266]: i was thinking that pointer to bool conversion been happening, was it  internal conversion to bool from std::optional  side ?

File: src/vtbackend/Terminal.cpp (hunk 2 above)
Comment (Yaraslaut) [reply src/vtbackend/Terminal.cpp:268]: I used it for debug what is happening, i was not sure whether we were waiting for response or were infinitely looping somewhere else, so i would like to have it as a log ouput, maybe we can move it in another log tag ?

File: src/vtbackend/Terminal.cpp (hunk 1 above)
Comment (Yaraslaut) [reply src/vtbackend/Terminal.cpp:266]: Oh, I see, indeed it has operator bool()

Comment (Yaraslaut) [reply src/vtbackend/Terminal.cpp]: Updated

**Pull request merged**